pkgs/development/python-modules/scipy/default.nix

```diff
@@ -95,22 +102,23 @@ buildPythonPackage {
       --replace-fail "pybind11>=2.12.0,<2.13.0" "pybind11>=2.12.0" \
   '';
 
-  nativeBuildInputs = [
-    cython
-    gfortran
-    meson-python
-    nukeReferences
-    pythran
-    pkg-config
-    wheel
-    setuptools
-  ] ++ lib.optionals stdenv.isDarwin [
-    # Minimal version required according to:
-    # https://github.com/scipy/scipy/blob/v1.14.0/scipy/meson.build#L185-L188
-    (xcbuild.override {
-      sdkVer = "13.3";
-    })
-  ];
+  build-system =
+    [
+      cython
+      gfortran
+      meson-python
+      nukeReferences
+      pythran
+      pkg-config
```

Comment:
Some of these belong in `nativeBuildInputs`. Also, `numpy` is missing.

Reply:
> Some of these belong in `nativeBuildInputs`. Also, `numpy` is missing.

The existence and not identicallity of both `nativeBuildInputs` and `build-system` is new to me. I am still surprised that you require adding to `build-system` packages that are already part of `dependencies`. I also don't think enforcing this is worth it. I plan to open a discourse thread about these policies you started to enforce lately, and that we discussed in the related numpy/scipy threads.

Reply:
https://github.com/NixOS/nixpkgs/pull/333221#discussion_r1737995289